### PR TITLE
Fix bbox computation for purposes, add test.

### PR DESCRIPTION
**Pull request by ppt-adsk**

### CODE REVIEW

File: lib/mayaUsd/ufe/UsdObject3d.cpp [hunk 1]
```diff
@@ -75,7 +75,9 @@ Ufe::BBox3d UsdObject3d::boundingBox() const
     // as the time.
 
     auto bbox = UsdGeomImageable(fPrim).ComputeUntransformedBound(
-        getTime(sceneItem()->path()), UsdGeomTokens->default_);
+        getTime(sceneItem()->path()), UsdGeomTokens->default_,
+        UsdGeomTokens->guide, UsdGeomTokens->proxy, UsdGeomTokens->render
```
Comment (kxl-adsk) [inline lib/mayaUsd/ufe/UsdObject3d.cpp:79]: On the proxy shape, we have toggles that control what will be rendered. I wonder if we shouldn't be respecting it here because purpose helps you manage complexity and including everything when computing bounding box may be simply too expensive.

File: lib/mayaUsd/ufe/Utils.cpp [hunk 2]
```diff
@@ -333,6 +334,54 @@ UsdTimeCode getTime(const Ufe::Path& path)
     return proxyShape->getTime();
 }
 
+TfTokenVector getProxyShapePurposes(const Ufe::Path& path)
```
Comment (ppt-adsk) [inline lib/mayaUsd/ufe/Utils.cpp:337]: Copy-pasted and adapted from getTime().  Arguably, getTime() itself has duplication from the stage map, and all calls to nameToDagPath() in the existing code base should be examined for performance and centralized.  This is out of scope for the current branch.

File: test/testUtils/testUtils.py [hunk 3]
```diff
@@ -30,6 +30,10 @@ def assertVectorAlmostEqual(testCase, a, b, places=7):
     for va, vb in zip(a, b):
         testCase.assertAlmostEqual(va, vb, places)
 
+def assertVectorNotAlmostEqual(testCase, a, b, places=7):
```
Comment (ppt-adsk) [inline test/testUtils/testUtils.py:33]: Thought I would need this, ended up not using it, but left it in, as it's useful.

**Review by kxl-adsk — APPROVED** [review]

**Review by seando-adsk — APPROVED** [review]
